cpp/src/io/parquet/reader_impl_chunking.cu

```diff
@@ -341,14 +379,715 @@ template <typename T = uint8_t>
 }
 
 struct row_count_compare {
-  __device__ bool operator()(cumulative_row_info const& a, cumulative_row_info const& b)
+  __device__ bool operator()(cumulative_page_info const& a, cumulative_page_info const& b)
   {
     return a.row_count < b.row_count;
   }
 };
 
+std::pair<size_t, size_t> get_row_group_size(RowGroup const& rg)
+{
+  auto compressed_size_iter = thrust::make_transform_iterator(
+    rg.columns.begin(), [](ColumnChunk const& c) { return c.meta_data.total_compressed_size; });
+
+  // the trick is that total temp space needed is tricky to know
+  auto const compressed_size =
+    std::reduce(compressed_size_iter, compressed_size_iter + rg.columns.size());
+  auto const total_size = compressed_size + rg.total_byte_size;
+  return {compressed_size, total_size};
```

Comment:
Right.  This is a remnant from early on where I was experimenting with how to determine how many compressed bytes should be loaded.  The state it's in now is pretty solid so this can be simplified.